### PR TITLE
Hierarchy driver

### PR DESCRIPTION
**Pull request by aprokop**
I would also like to add parameter reading from a command line to override the provided input deck parameters.

### CODE REVIEW

**Comment by aprokop** [comment]
Interesting, I have no clue why this fails.

**Comment by Rombur** [comment]
Just close and reopen the PR. The laplace tests are not great because there can be some race conditions when removing the output files.

**Comment by aprokop** [comment]
Is is possible to `fsync` after writing data, and then `barrier` to avoid race conditions?

**Comment by aprokop** [comment]
@Rombur OK. Updated this PR with the fix.

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -1,12 +1,15 @@
 INCLUDE(${CMAKE_SOURCE_DIR}/cmake/SetupTests.cmake)
 
+INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR})
+
```
Comment (Rombur) [inline tests/CMakeLists.txt]: Why do you do this?

File: tests/test_hierarchy.cc [hunk 2]
```diff
@@ -18,6 +18,8 @@
 #include <mfmg/dealii_adapters.hpp>
 #include <mfmg/hierarchy.hpp>
 
+#include <test_hierarchy_helpers.hpp>
+
```
Comment (Rombur) [inline tests/test_hierarchy.cc]: Is it the reason why you want to include the `tests` directory in CMake? You can just use `#include "test_hierarchy_helpers.hpp"` instead. That's what we do with `laplace.hpp`

File: tests/test_hierarchy_driver.cc [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    res[i + 1] = rel_residual;
+  }
+
+  double const conv_rate = res[n_cycles] / res[n_cycles - 1];
+  pcout << "Convergence rate: " << std::fixed << std::setprecision(2)
+        << conv_rate << std::endl;
+}
+
+int main(int argc, char *argv[])
+{
```
Comment (Rombur) [inline tests/test_hierarchy_driver.cc]: Is this file a test or not? If it is not then don't call the file `test_hierarchy_driver` and if it is, then you need to add a test. Same with `test_hierarchy_driver_helper`

**Comment by aprokop** [comment]
@Rombur Addressed your comments.

**Pull request merged**